### PR TITLE
Add preferred facility alert to the 10-10EZR

### PR DESCRIPTION
**Pull request by longmd**
## Summary
Due to the 10-10EZ app sharing the same downstream submission API with the 10-10EZR, it was discovered that users may be enrolled in health benefits, but not have a preferred facility on file. This causes a validation issue with the API, as the facility code is a required field, but that value isn't collected on the EZR app. 

In order to alleviate this validation issue, this PR checks for the preferred facility value and blocks the user from accessing the app if that value is not set. This PR also implements GA event tracking to allow the team to record the volume of users who receive this alert.

## Related issue(s)
department-of-veterans-affairs/va.gov-team#72849

## Testing done

- [x] Unit testing was added to account for data sets with and without the preferred facility value

## Screenshots
Screenshot 2024-01-08 at 12 10 02 PM

## Acceptance criteria

- Users who do not have a preferred facility value are blocked from using the online form

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### CODE REVIEW

File: src/applications/ezr/tests/unit/components/IntroductionPage/SaveInProgressInfo.unit.spec.js [hunk 1]
```diff
@@ -183,6 +185,51 @@ describe('ezr <SaveInProgressInfo>', () => {
           expect(selector).to.contain.text(content['alert-server-title']);
         });
       });
+
+      context('when the user has a preferred facility on file', () => {
+        it('should not render preferred facility alert', () => {
+          const { props, mockStore } = getData({
+            showLoader: false,
+            loggedIn: true,
+            loaState: 3,
```
Comment (mkerns1) [inline src/applications/ezr/tests/unit/components/IntroductionPage/SaveInProgressInfo.unit.spec.js:194]: Just for my clarity is this intentionally called 'loaState'? Or could this be a typo for 'loadState'?

File: src/applications/ezr/tests/unit/components/IntroductionPage/SaveInProgressInfo.unit.spec.js [hunk 2]
```diff
@@ -183,6 +185,51 @@ describe('ezr <SaveInProgressInfo>', () => {
...
+      });
+
+      context(
+        'when the user does not have a preferred facility on file',
+        () => {
+          it('should render preferred facility alert', () => {
+            const { props, mockStore } = getData({
+              showLoader: false,
+              loggedIn: true,
+              loaState: 3,
```
Comment (mkerns1) [inline src/applications/ezr/tests/unit/components/IntroductionPage/SaveInProgressInfo.unit.spec.js:216]: I also see it here which makes my believe it was intentional but I just want to confirm and maybe hear the reasoning for the naming convention for my reference.

**Review by mkerns1 — APPROVED** [review]
Thanks for the clarification Matt. LGTM!

File: src/applications/ezr/tests/unit/components/IntroductionPage/SaveInProgressInfo.unit.spec.js (hunk 1 above)
Comment (longmd) [reply src/applications/ezr/tests/unit/components/IntroductionPage/SaveInProgressInfo.unit.spec.js:194]: LOA is “level of assurance” and it relates to the verification of the user account. So if you see a user that has an LOA status of 1, that means they have created an account, but they have not verified their identity. This restricts most of the functions you can perform as a logged in user (such as filling out our forms and other forms, editing personal info, etc). Users with an LOA status of 3 have fully verified their identity and are authorized to fill out forms and update their profile data.

**Pull request merged**